### ISSUE TITLE
[TW-737] SOLARIS not defined by CMake

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#772]
_Owen Clarke on 2011-06-28T01:54:20Z says:_ 

 Running "cmake ." doesn't properly detect a Solaris machine and puts "#define UNKNOWN" in cmake.h

Attached is a patch for CMakeLists.txt that fixes this for Oracle Solaris, I can test on OpenIndiana as well but haven't yet.

It also appears that the other possible systems wouldn't get defined by CMake either: CYGWIN, OPENBSD, HAIKU, WIN32; but I am not familiar with CMake or with the taskwarrior codebase enough to know if this is correct or matters.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2011-06-28T01:54:20Z
Modified: 2014-02-09T02:06:35Z
```

**Comment by taskwarrior** [comment]
_Owen Clarke on 2011-06-28T01:55:39Z says:_ 

 The patch is against code from the 2.0.0 branch as of yesterday, but I had similar problems with cmake in 1.9.4.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-06-28T01:56:54Z says:_ 

 Thank you Owen - I'm going to address this immediately.  Thank you very much for pointing this out.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-06-28T02:06:44Z says:_ 

 Perfect - thank you Owen, it's pushed and you are credited.

Please note: If you build the 2.0.0 branch, there are about 700 failing unit tests right now (http://tasktools.org/tinderbox/task.html).  I'm dogfooding 2.0, but then I know what's broken.  Not recommended.  Back up you data.  Keep a fire extinguisher handy.

**Comment by taskwarrior** [comment]
_Owen Clarke on 2011-06-28T02:14:09Z says:_ 

 Paul Beckingham wrote:
> Perfect - thank you Owen, it's pushed and you are credited.
> 
> Please note: If you build the 2.0.0 branch, there are about 700 failing unit tests right now (http://tasktools.org/tinderbox/task.html).  I'm dogfooding 2.0, but then I know what's broken.  Not recommended.  Back up you data.  Keep a fire extinguisher handy.

Thanks, I've noticed that a lot of things don't work and whole cmd implementations are commented out; I thought I'd just try 2.0.0 out and get it to compile on Solaris.

**Comment by taskwarrior** [comment]
_Paul Beckingham on 2011-06-28T02:28:14Z says:_ 

 > Thanks, I've noticed that a lot of things don't work and whole cmd implementations are commented out; I thought I'd just try 2.0.0 out and get it to compile on Solaris.

First, thank you for being vigilant about Solaris - we want the platform to be supported, but don't use it ourselves, as you already know.

Second, you're right - the whole filtering mechanism was replaced with something better, but it's sufficiently different that I had to comment out all the commands, just so I could get one to compile (CmdCustom.cpp - basically all tabular reports).  So there is much that is broken at the moment, but this is temporary, and I'm restoring functionality bit by bit.  It will be worth it.
